### PR TITLE
[12.0][MIG] stock_inventory_barcode

### PR DESCRIPTION
**Pull request by alexis-via**
Migration from v10 to v12.
Several improvements over the v10 version:
- handle multiple stock locations (while remaining simple when you only have 1)
- handle products tracked by lot and serial number

### CODE REVIEW

File: stock_inventory_barcode/wizard/stock_inventory_barcode.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            if len(products) == 1:
+                product = products[0]
+                self.product_id = product
+                self.product_tracking = product.tracking
+                self.uom_id = product.uom_id
+            elif len(products) > 1:
+                return {'warning': {
+                    'title': _('Error'),
+                    'message': _(
+                        'Several stockable products have been found '
```
Comment (kevinkhao) [inline stock_inventory_barcode/wizard/stock_inventory_barcode.py:91]: Non-blocking, seems redundant with already-existsing barcode unique sql contraint https://github.com/odoo/odoo/blob/57f078b68b79bfdb53d4d7bb71b6672b4ad8e661/addons/product/models/product.py#L158

File: stock_inventory_barcode/wizard/stock_inventory_barcode.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        'stock.location', string='Location', required=True)
+    multi_stock_location = fields.Boolean(readonly=True)
+    lot_id = fields.Many2one(
+        'stock.production.lot', string='Lot')
+    product_tracking = fields.Selection([
+        ('serial', 'By Unique Serial Number'),
+        ('lot', 'By Lots'),
+        ('none', 'No Tracking'),
+        ], string="Tracking", required=True)
+    note = fields.Text()
```
Comment (kevinkhao) [inline stock_inventory_barcode/wizard/stock_inventory_barcode.py:39]:
```suggestion
    note = fields.Text(readonly=True)
```

**Review by kevinkhao — APPROVED** [review]
Functional tests LGTM. Functionality and onchanges seems to work as intended. Tiny non-blocking suggestions.

**Review by dsolanki-initos — CHANGES_REQUESTED** [review]
We need to follow the guideline to commit, Can you squash all the commits in : Migration to 12.0

**Comment by dsolanki-initos** [comment]
Can you make all checks green?

**Comment by dsolanki-initos** [comment]
Can you make all checks green?

**Comment by dsolanki-initos** [comment]
It would be nice if you try to make runbot happy.

**Comment by dsolanki-initos** [comment]
@chienandalu it would be nice if you add reviews here.

**Comment by chienandalu** [comment]
> @chienandalu it would be nice if you add reviews here.

Sorry, I don't know this module.

**Comment by dsolanki-initos** [comment]
@rousseldenis Could you review this pull?

**Comment by dsolanki-initos** [comment]
Can we merge this PR?

**Comment by dsolanki-initos** [comment]
Can we merge this PR?

**Comment by dsolanki-initos** [comment]
Can we merge this PR?

**Comment by dsolanki-initos** [comment]
@dreispt Can we merge this PR?

**Comment by dreispt** [comment]
@dsolanki-initos Your review is requesting changes 😉 
I guess that's outdated.

/ocabot merge nobump

**Pull request merged**